File: scripts/execute-migration.js
#!/usr/bin/env node

/**
 * Execute Supabase SQL migration directly via API
 * Usage: node scripts/execute-migration.js
 */

const fs = require('fs');
const path = require('path');

const SUPABASE_URL = process.env.VITE_SUPABASE_URL;
const SUPABASE_SERVICE_ROLE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!SUPABASE_URL || !SUPABASE_SERVICE_ROLE_KEY) {
  console.error('❌ Missing SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY in environment');
  console.error('   Please add SUPABASE_SERVICE_ROLE_KEY to .env.local');
  process.exit(1);
}

async function executeMigration() {
  const migrationPath = path.join(
    __dirname,
    '../supabase/migrations/create_shippers_and_operational_units.sql'
  );

  if (!fs.existsSync(migrationPath)) {
    console.error(`❌ Migration file not found: ${migrationPath}`);
    process.exit(1);
  }

  const sql = fs.readFileSync(migrationPath, 'utf-8');

  // Split by semicolons and filter empty statements
  const statements = sql
    .split(';')
    .map((s) => s.trim())
    .filter((s) => s.length > 0);

  console.log(`📋 Found ${statements.length} SQL statements to execute`);

  for (let i = 0; i < statements.length; i++) {
    const statement = statements[i];
    console.log(`\n[${i + 1}/${statements.length}] Executing statement...`);

    try {
      const response = await fetch(`${SUPABASE_URL}/rest/v1/rpc/sql_exec`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${SUPABASE_SERVICE_ROLE_KEY}`,
          'X-Client-Info': 'supabase-migration/1.0',
        },
        body: JSON.stringify({ query: statement + ';' }),
      });

      if (!response.ok) {
        const error = await response.text();
        console.warn(`⚠️  Statement ${i + 1} might have failed or already exists:`);
        console.warn(`   Status: ${response.status}`);
        console.warn(`   Response: ${error.substring(0, 200)}`);
      } else {
        console.log(`✅ Statement ${i + 1} executed`);
      }
    } catch (error) {
      console.warn(`⚠️  Could not execute statement ${i + 1}:`, error.message);
    }
  }

  console.log('\n✅ Migration execution completed!');
  console.log('   If any statements failed, please execute the SQL manually in Supabase Dashboard.');
}

executeMigration().catch((error) => {
  console.error('❌ Migration failed:', error);
  process.exit(1);
});

File: scripts/execute_migration.py
#!/usr/bin/env python3

"""
Execute Supabase SQL migration using Python
Requires: pip install supabase
"""

import os
import sys
from pathlib import Path

def main():
    # Load environment
    env_file = Path('.env.local')
    if not env_file.exists():
        print('❌ .env.local not found')
        sys.exit(1)

    env_vars = {}
    with open(env_file) as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith('#'):
                if '=' in line:
                    key, val = line.split('=', 1)
                    env_vars[key.strip()] = val.strip()

    supabase_url = env_vars.get('VITE_SUPABASE_URL')
    supabase_anon_key = env_vars.get('VITE_SUPABASE_ANON_KEY')

    if not supabase_url or not supabase_anon_key:
        print('❌ Missing VITE_SUPABASE_URL or VITE_SUPABASE_ANON_KEY in .env.local')
        sys.exit(1)

    # Read migration file
    migration_file = Path('supabase/migrations/create_shippers_and_operational_units.sql')
    if not migration_file.exists():
        print(f'❌ Migration file not found: {migration_file}')
        sys.exit(1)

    sql = migration_file.read_text()

    print('📋 Executing SQL migration via Supabase...')
    print(f'   URL: {supabase_url}')
    print()

    try:
        from supabase import create_client

        # Initialize client with anon key (read-only)
        client = create_client(supabase_url, supabase_anon_key)

        # Try to execute SQL via rpc if available
        # This will likely fail without proper setup, so we'll catch and inform
        result = client.rpc('sql_exec', {'query': sql}).execute()
        print('✅ Migration executed successfully!')

    except ImportError:
        print('❌ Supabase Python client not installed')
        print('   Install with: pip install supabase')
        print()
        print('⚠️  Alternatively, execute the migration manually:')
        print('   1. Open: https://supabase.com/dashboard')
        print('   2. Go to SQL Editor')
        print('   3. Paste the contents of: supabase/migrations/create_shippers_and_operational_units.sql')
        print('   4. Click Run')
        sys.exit(1)

    except Exception as e:
        print(f'⚠️  Could not execute migration via API: {str(e)}')
        print()
        print('Please execute the migration manually:')
        print('   1. Open: https://supabase.com/dashboard')
        print('   2. Go to SQL Editor')
        print('   3. Paste the contents of: supabase/migrations/create_shippers_and_operational_units.sql')
        print('   4. Click Run')
        sys.exit(1)

if __name__ == '__main__':
    main()
